java/arcs/core/storage/api/Entity.kt

```diff
@@ -35,3 +38,21 @@ interface EntitySpec<T : Entity> {
      */
     fun deserialize(data: RawEntity): T
 }
+
+/**
+ * Try to extract the primitive value from a [ReferencablePrimitive].
+ */
+@Suppress("UNCHECKED_CAST")
+fun <T : Any> toPrimitiveValue(
```

Comment:
Optional: you could make this an extension function on Referenceable:

```
myReferencable.toPrimitiveValue(String::class)
```

I think having a default might be kind of dangerous, it's probably better to throw an exception.

Reply:
I thought of that (extension), but when I tried to define it, it didn't work because RawEntity returns Referencable? and my Java-esqe brain didn't grok 'null.extensionFunction()' (even though they work just like in GWT).  But I see you can define 'Type?.extensionFunction()' and it works. :)

I made it throw an exception when the type is wrong, but not when the data is null, because the existing code assumes that null data for a primitive Entity field is the default for that type. (that's what Entity's default constructor defaults to)